File: Problems/Algorithms/509. Fibonacci Number/fibonacci_number.py
#!/usr/bin/env python3

from functools import lru_cache


class Solution:
    @lru_cache(maxsize=None)
    def fib(self, n: int) -> int:
        if n < 2:
            return n
        return self.fib(n-1) + self.fib(n-2)

File: Problems/Algorithms/509. Fibonacci Number/FibonacciNumber.kt
class Solution {
    fun fib(n: Int): Int {
        if (n == 0) return 0
        if (n == 1) return 1

        val dp: IntArray = IntArray(n+1)
        dp[1] = 1
        for (i in 2..n) {
            dp[i] = dp[i-1] + dp[i-2]
        }
        return dp[n]
    }
}
